examples/usages/context_menu.rs

```diff
@@ -0,0 +1,201 @@
+//! This example illustrates how to create a context menu that changes the clear color
+
+use bevy::{
+    color::palettes::basic,
+    ecs::{relationship::RelatedSpawner, spawn::SpawnWith},
+    prelude::*,
+};
+use std::fmt::Debug;
+
+/// event opening a new context menu at position `pos`
+#[derive(Event)]
+struct OpenContextMenu {
+    pos: Vec2,
+}
+
+/// event will be send to close currently open context menus
+#[derive(Event)]
+struct CloseContextMenus;
+
+/// marker component identifying root of a context menu
+#[derive(Component)]
+struct ContextMenu;
+
+/// context menu item data storing what background color `Srgba` it activates
+#[derive(Component)]
+struct ContextMenuItem(Srgba);
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_observer(on_trigger_menu)
+        .add_observer(on_trigger_close_menus)
+        .add_observer(create_hover_observer::<Out>(basic::WHITE))
```

Comment:
Maybe something like `text_color_on_hover` would be more descriptive?

Reply:
good one